Documentation/gettingstarted_kafka.rst

```diff
@@ -0,0 +1,392 @@
+indicated by "app: kafka"
+in the "endpointSelector" section).  The rule will apply to connections from pods with label
+"app: empire-outpost" as indicated by the "fromEndpoints" section.   The rule explicitly matches
+Kafka connections destined to TCP 9092, and white-lists a list of protocol requests according to various
+fields found in Kafka protocol messages.  Specifically, each request contains an "apiKey" field that
+indicates the type of request, and we white-list several apiKey's here, the most important being the
+apiKey "fetch",
+which allows the client to consume from a particular topic.   Notice that requests with "apiKey: fetch"
+are further limited by a "topic" field, in this case "empire-announce".
+
+The full policy adds two additional rules that permit the legitimate produce request
+(topic "empire-announce" and topic "deathstar-plans") from *empire-hq* and the
+legitimate fetch requests (topic = "deathstar-plans") from *empire-backup*.  The full policy
+can be reviewed by opening the URL in the command below in a browser.
+
+Apply this Kafka-aware network security policy using ``kubectl`` in the main window:
+
+.. parsed-literal::
+
+    $ kubectl create -f \ |SCM_WEB|\/examples/kubernetes-kafka/kafka-sw-security-policy.yaml
+
+If we then again try to produce a message from outpost-9999 to *empire-annnounce*, it is denied.
+Type control-c and then run:
+
+::
+
+  $ echo "Vader Trips on His Own Cape" | ./kafka-produce.sh --topic empire-announce
+  [2017-10-31 07:08:34,088] ERROR Error when sending message to topic empire-announce with key: null, value: 33  bytes with error: (org.apache.kafka.clients.producer.internals.ErrorLoggingCallback)
+  org.apache.kafka.common.errors.TopicAuthorizationException: Not authorized to access topics: [empire-announce]
+
+This is because the policy does not allow messages with apiKey = "produce" for topic "empire-announce" from
+containers with label app = empire-outpost.  Its worth noting that we don't simply drop the message (which
+could easily be confused with a network error), but rather we respond with the Kafka access denied error
+(similar to how HTTP would return an error code of 403 unauthorized).
+
+Likewise, if the outpost container ever tries to consume from topic *deathstar-plans*, it is denied, as
+apiKey = fetch is only allowed for topic *empire-announce*.
+
+To test, from the outpost-9999 terminal, run:
+
+::
+
+  $ ./kafka-consume.sh --topic deathstar-plans
+  [2017-10-31 07:09:36,679] WARN Not authorized to read from topic deathstar-plans. (org.apache.kafka.clients.consumer.internals.Fetcher)
+  [2017-10-31 07:09:36,683] ERROR Error processing message, terminating consumer process:  (kafka.tools.ConsoleConsumer$)
+  org.apache.kafka.common.errors.TopicAuthorizationException: Not authorized to access topics: [deathstar-plans]
+  Processed a total of 0 messages
+
+This is blocked as well.  Imagine how different things would have been if the empire had been using
```

Comment:
Potentially add to first sentence: "This is blocked as well, thanks to the Cilium network policy."